backend/config/corpora-api.yml

```diff
@@ -418,11 +418,15 @@ paths:
       tags:
         - collections
       summary: Get a collection's full details
+      security:
+        - cxguserCookie: []
+        - dummyAuth: []
       description: >-
         This will return all datasets and associated attributes.
       operationId: corpora.lambdas.api.v1.collection.get_collection_details
       parameters:
         - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/query_visibility"
```

Comment:
For clarification: Is `query_visibility` expecting FE to pass a param to tell BE if the collection is private or not? If so, given that FE doesn't know if a collection is private or public, BE might need to look this up in the database?

Reply:
we may need to update list collections to include the visibility and retrieve the users collections if they are logged in.

Reply:
In that case, would FE still need to pass `visibility` for the collection?

I'm thinking about the case that a user bookmarks a private collection URL `/collections/private-collection-id` and then hit that URL directly, then the FE won't have a chance to get the collection's visibility. So it doesn't seem like FE is a reliable source of truth for knowing whether a collection is public or private

Another question is, if a malicious user passed `visibility = Public` for a private collection, would they gain access to the private collection?

Reply:
> Another question is, if a malicious user passed visibility = Public for a private collection, would they gain access to the private collection?

N, they would need to be the owner or use the obfuscated id to be able read.

Reply:
> I'm thinking about the case that a user bookmarks a private collection URL /collections/private-collection-id and then hit that URL directly, then the FE won't have a chance to get the collection's visibility. So it doesn't seem like FE is a reliable source of truth for knowing whether a collection is public or private

You mean they save the API request. Since the API is not public, we don't guarantee anything if they use the API directly.
Does the frontend URL need to mirror the backend API request format? Couldn't the front end try to get the public and if that fails. try to get the private, or something to that effect.